File: app/models/document.rb
class Document < ActiveRecord::Base
  belongs_to :user
  has_many :assemblies

  attr_accessible :image_url, :cover_image, :cover_image_url, :retained_cover_image, :remove_cover_image, :cover_image_name, :cover_image_uid
  attr_accessible :public, :title, :slug, :footer, :header, :main, :primary_color, :secondary_color, :assembly_id

  image_accessor :cover_image

  attr_accessor :image_url
end

File: app/assets/javascripts/models/document.js
App.Document = DS.Model.extend({
  slug: DS.attr('string'),
  title: DS.attr('string'),
  imageUrl: DS.attr('string'),
  public: DS.attr('boolean'),

  footer: DS.attr('string'),
  header: DS.attr('string'),
  main: DS.attr('string'),
  primaryColor: DS.attr('string'),
  secondaryColor: DS.attr('string'),
  
  printUrl: function() {
    return "/documents/" + this.get('id') + ".pdf"
  }.property(),
  colorChanged: function(e, attr) {
    var value = this.get(attr);
    var oppValue = (attr === 'primaryColor') ? this.get('secondaryColor') : this.get('primaryColor');
    var oppAttr = (attr === 'primaryColor') ? 'secondaryColor' : 'primaryColor';

    if( value !== 'white' ) {
      this.set(oppAttr, 'white');
    }

  }.observes('primaryColor', 'secondaryColor'),

  // user: DS.attr('App.User'),
  assembly: DS.belongsTo('App.Assembly')
});